### ISSUE TITLE
UI left in inconsistent state after SSH remote connection fails

### ISSUE THREAD

**Issue by samuela** [microsoft/vscode#92876]
Issue Type: <b>Bug</b>

1. Try to connect a remote host (named "deirdre" here)
2. That fails, but does so mostly silently. There are two notification bubbles in the Notifications pane, but these aren't shown by the time the SSH vscode window appears.
3. After it fails to connect, the bottom right green indicator still makes it look as though we are connected. Clicking on "Open folder..." shows files/folders on the local machine. This is juxtaposed with a button saying "Show Local" which opens up the macOS file picker dialog as if we are not already looking at local files.

![Screen Shot 2020-03-17 at 12 04 53 PM](https://user-images.githubusercontent.com/226872/76892756-6cc4eb00-6848-11ea-9aa7-e5573088d655.png)

VS Code version: Code - Insiders 1.44.0-insider (b7bfce30be3c30aa2c33fbe94fe82b81d4105668, 2020-03-17T05:39:17.564Z)
OS version: Darwin x64 19.2.0
Fetching remote diagnostics for 'SSH: deirdre' failed: Time limit reached

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz (4 x 2300)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|8.00GB (0.12GB free)|
|Process Argv|-psn_0_1179936|
|Screen Reader|no|
|VM|0%|

Fetching remote diagnostics for 'SSH: deirdre' failed: Time limit reached
</details>Extensions: none

**Comment by alexdima** [comment]
@samuela Does this reproduce every time for you? Do you just make up a random host name or how do you reproduce?

@roblourens From looking at the error message -- "Time limit reached" -- this indicates to me that the SSH extension has resolved the remote authority, but then VS Code cannot actually connect (open a socket) to the resolved `host:port`. Do you know how / why this would happen?

**Comment by roblourens** [comment]
Yeah, if the port is wrong for some reason, or the remote server crashes a second after it's launched, or if something is blocking the port forwarding, like if TCP port forwarding is disabled for the ssh server.

The log in the Remote-SSH output channel might give a clue.
